src/containers/Tenant/Diagnostics/TopQueries/TopQueries.tsx

```diff
@@ -56,23 +69,60 @@ export const TopQueries = ({path, type, changeSchemaTab}: TopQueriesProps) => {
         wasLoaded,
         error,
         data: {result: data = undefined} = {},
+        filters: storeFilters,
     } = useTypedSelector((state) => state.executeTopQueries);
 
+    const preventUpdate = useRef(false);
+
+    const [filters, setFilters] = useState<ITopQueriesFilters>(storeFilters);
+
     useAutofetcher(
-        () => dispatch(fetchTopQueries({database: path})),
-        [dispatch, path],
+        (isBackground) => {
+            if (preventUpdate.current) {
+                preventUpdate.current = false;
+                return;
+            }
+
+            if (!isBackground) {
+                dispatch(
+                    setTopQueriesState({
+                        wasLoaded: false,
+                        data: undefined,
+                    }),
+                );
+            }
+
+            // @ts-expect-error
+            // typed dispatch required, remove error expectation after adding it
+            dispatch(fetchTopQueries({database: path, filters})).then(
```

Comment:
Just a suggest - maybe it would be more handy to move this handler into separate function.

Reply:
refactored it a bit, please see the [compare](https://github.com/ydb-platform/ydb-embedded-ui/compare/2be59564a45858c0ff70077d94ea46c93c036b49..a8c8df29073c4f63485eae319dd8f4d27f7ea362)